File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.ts
function minSubArrayLen(target: number, nums: number[]): number {
    let l = 0;
    let Min = Infinity;
    let curSum = 0;
    for(let r = 0; r < nums.length; r ++){
        curSum += nums[r];
        while(curSum >= target){
            Min = Math.min(Min,r - l + 1);
            curSum -= nums[l];
            l ++;
        }
    }
    return Min === Infinity ? 0 : Min;
};

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int l = 0;
        int curSum = 0;
        int min_len = Integer.MAX_VALUE;
        for(int r = 0; r < nums.length; r ++) {
            curSum += nums[r];
            while(curSum >= target){
                min_len = Math.min(min_len , r - l + 1);
                curSum -= nums[l];
                l ++;
            }
        }
        return min_len == Integer.MAX_VALUE ? 0 : min_len;
    }
}